@AH-64D Apache Longbow/Addons/fza_ah64_ihadss/functions/fn_ihadssDraw.sqf

```diff
@@ -138,24 +139,14 @@ if (cameraView == "GUNNER" && player == gunner _heli && !_powerOnState) then {
 };
 
 //IHADSS INIT
-if (!(_heli getVariable "fza_ah64_monocleinbox") || _powerOnState && !(_heli getVariable "fza_ah64_monocleinbox") || !(_heli getVariable "fza_ah64_monocleinbox")) then {
-    if (isNil "fza_ah64_ihadssinit") then {
-        1 cutrsc["fza_ah64_raddisp", "PLAIN", 0.01, false];
-        ((uiNameSpace getVariable "fza_ah64_raddisp") displayCtrl 130) ctrlSetText "\fza_ah64_US\tex\HDU\ihadss.paa";
-
-        _ihadssidx = 121;
-
-        while {
-            (_ihadssidx < 206)
-        }
-        do {
-            ((uiNameSpace getVariable "fza_ah64_raddisp") displayCtrl _ihadssidx) ctrlSetTextColor fza_ah64_hducolor;
-            _ihadssidx = _ihadssidx + 1;
-        };
+if !(_heli getvariable "fza_ah64_ihadssinit") then {
+    1 cutrsc["fza_ah64_raddisp", "PLAIN", 0.01, false];
+    ((uiNameSpace getVariable "fza_ah64_raddisp") displayCtrl 130) ctrlSetText "\fza_ah64_US\tex\HDU\ihadss.paa";
 
-        _rocketcode = "???";
-        fza_ah64_ihadssinit = true;
+    for "_i" from 121 to 206 do {
+        ((uiNameSpace getVariable "fza_ah64_raddisp") displayCtrl _i) ctrlSetTextColor _hduColour;
     };
+    _heli setVariable ["fza_ah64_ihadssinit", true];
```

Comment:
I am not sure the scope of this variable should be attached to the heli rather than global, or as a variable attached to the display

Reply:
it would work either way